typeinfo.go

```diff
@@ -357,19 +362,37 @@ func getFields(typ reflect.Type) *fields {
 		if name == "-" {
 			continue
 		}
+		ftyp := f.Type
+		if ftyp.Kind() == reflect.Ptr {
+			ftyp = ftyp.Elem()
+		}
+		if opts.Contains("inline") && ftyp.Kind() == reflect.Struct {
+			flds := getFields(ftyp)
+			for name, field := range flds.Fields {
+				if _, ok := fs.Fields[name]; ok {
+					// Don't overwrite shadowed fields.
+					continue
+				}
+				field.index = append(f.Index, field.index...)
+				fs.add(name, field)
+			}
+			// If the struct has no (exported) fields try to encode/decode it as
+			// a whole later.
+			if len(flds.Fields) != 0 {
+				continue
+			}
```

Comment:
I find this block a bit confusing to understand. Probably:

```
inlinedFields := getFields(ftyp)
if len(inlinedFields) > 0 {
    inlineFields(fs, inlinedFields)
    continue
}
```

What do you think?